Monocle/Auth/Type.dhall

```diff
@@ -0,0 +1 @@
+{ name : Text, enforce_auth : Optional Bool, provider : ./Provider.dhall }
```

Comment:
Thinking about the future to support extra provider, I think we could have:
```dhall
{ provider : ./Provider.dhall, extra_providers : List ./Provider.dhall }
```
Which would be quite similar to a NonEmpty List, that's good.

Though, perhaps we should move the `name` attribute inside the provider.